batcher/aligned-batcher/src/retry/mod.rs

```diff
@@ -0,0 +1,53 @@
+pub mod batcher_retryables;
+
+use backon::ExponentialBuilder;
+use backon::Retryable;
+use std::{future::Future, time::Duration};
+
+#[derive(Debug)]
+pub enum RetryError<E> {
+    Transient(E),
+    Permanent(E),
+}
+
+impl<E: std::fmt::Display> std::fmt::Display for RetryError<E> {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            RetryError::Transient(e) => write!(f, "{}", e),
+            RetryError::Permanent(e) => write!(f, "{}", e),
+        }
+    }
+}
+
+impl<E> RetryError<E> {
+    pub fn inner(self) -> E {
+        match self {
+            RetryError::Transient(e) => e,
+            RetryError::Permanent(e) => e,
+        }
+    }
+}
+
+impl<E: std::fmt::Display> std::error::Error for RetryError<E> where E: std::fmt::Debug {}
+
+pub async fn retry_function<FutureFn, Fut, T, E>(
```

Comment:
Since the implementation of exp backoff is different if the fn is async or blocking, I'd specify which type we are covering (async).
Also I'd add refference to the library https://docs.rs/backon/latest/backon/#retry-an-async-function

Reply:
Added!